src/auth.rs

```diff
@@ -62,29 +56,21 @@ impl TidalCredentials {
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
-pub struct SessionInfo {
-    pub user_id: Option<u32>,
-    pub session_id: Option<String>,
+pub struct Session {
+    pub user_id: u32,
+    pub session_id: String,
     pub country_code: String,
 }
 
-impl SessionInfo {
-    pub fn new() -> Self {
-        Self {
-            user_id: None,
-            session_id: None,
-            country_code: String::new(),
-        }
-    }
-
-    pub async fn get_session(&self, token: &str, username: &str, password: &str) -> Option<Self> {
+impl Session {
+    pub async fn get_session(token: &str, username: &str, password: &str) -> Option<Self> {
         let mut payload: HashMap<&str, &str> = HashMap::new();
         payload.insert("username", username);
         payload.insert("password", password);
-        self.fetch_session_data(token, &payload).await
+        Self::fetch_session_data(token, &payload).await
```

Comment:
do you need `Self::` at all if it's part of the same `impl` block?

Reply:
@jpalvarezl  yes it's needed, as it's calling method implemented in the Struct, not in the module.

you would get this error if you remove the `Self`:

```
   |
70 |         fetch_session_data(token, &payload).await
   |         ^^^^^^^^^^^^^^^^^^ not found in this scope
```